src/dotnet/APIView/APIViewWeb/Managers/SamplesRevisionsManager.cs

```diff
@@ -36,11 +39,34 @@ public SamplesRevisionsManager(
             _commentsManager = commentManager;
         }
 
+        public async Task<SamplesRevisionModel> GetSamplesRevisionAsync(string reviewId, string samplesRevisionId)
+        {
+            return await _samplesRevisionsRepository.GetSamplesRevisionAsync(reviewId, samplesRevisionId);
+        }
+
         public async Task<IEnumerable<SamplesRevisionModel>> GetSamplesRevisionsAsync(string reviewId)
         {
             return await _samplesRevisionsRepository.GetSamplesRevisionsAsync(reviewId);
         }
 
+        /// <summary>
+        /// Retrieve Revisions from the Revisions container in CosmosDb after applying filter to the query.
+        /// </summary>
+        /// <param name="user"></param>
+        /// <param name="pageParams"></param> Contains pagination info
+        /// <param name="filterAndSortParams"></param> Contains filter and sort parameters
+        /// <returns></returns>
+        public async Task<PagedList<SamplesRevisionModel>> GetSamplesRevisionsAsync(ClaimsPrincipal user, PageParams pageParams, FilterAndSortParams filterAndSortParams)
+        {
+            return await _samplesRevisionsRepository.GetSamplesRevisionsAsync(user, pageParams, filterAndSortParams);
+        }
+
+        public async Task<SamplesRevisionModel> GetLatestSampleRevisionsAsync(string reviewId)
+        {
+            var samplesRevisions = await _samplesRevisionsRepository.GetSamplesRevisionsAsync(reviewId);
+            return samplesRevisions.OrderByDescending(r => r.CreatedOn).FirstOrDefault();
```

Comment:
This will throw exception if there are no samples under that review ID. It's still better to handle that case here even if it is handled in the caller to avoid any issue in the future.